packages/expo-updates/ios/EXUpdates/AppLoader/FileDownloader.swift

```diff
@@ -108,8 +148,65 @@ public final class FileDownloader {
         return
       }
 
+      let httpResponse = response as? HTTPURLResponse
+      let isDiffResponse = httpResponse.map(Self.isDiffResponse) ?? false
+
+      if isDiffResponse && !allowPatch {
+        let underlyingError = NSError(
+          domain: "EXUpdatesFileDownloader",
+          code: 1001,
+          userInfo: [NSLocalizedDescriptionKey: "Received Hermes diff when diff support disabled"]
+        )
+        let cause = UpdatesError.fileDownloaderUnknownError(cause: underlyingError)
+        self.logger.warn(
+          message: "Received Hermes diff when diff support disabled for asset \(asset.key ?? asset.filename)",
+          code: UpdatesErrorCode.assetsFailedToLoad,
+          updateId: extraHeaders[Self.ExpoRequestedUpdateIdHeader] as? String,
+          assetId: asset.key ?? asset.filename
+        )
+        self.logger.error(cause: cause, code: UpdatesErrorCode.assetsFailedToLoad)
+        errorBlock(cause)
+        return
+      }
+
+      if allowPatch,
+        let response = httpResponse,
+        isDiffResponse {
+        do {
+          let (patchedData, hashBase64String) = try self.applyHermesDiff(
+            asset: asset,
+            diffData: data,
+            destinationPath: destinationPath,
+            extraHeaders: extraHeaders,
+            expectedBase64URLEncodedSHA256Hash: expectedBase64URLEncodedSHA256Hash
+          )
+          successBlock(patchedData, response, hashBase64String)
+          return
+        } catch {
+          let updatesError = UpdatesError.fileDownloaderUnknownError(cause: error)
+          self.logger.error(
+            cause: updatesError,
+            code: UpdatesErrorCode.assetsFailedToLoad,
+            updateId: extraHeaders[Self.ExpoRequestedUpdateIdHeader] as? String,
+            assetId: asset.key ?? asset.filename
+          )
+          self.downloadAsset(
+            asset: asset,
+            fromURL: url,
+            verifyingHash: expectedBase64URLEncodedSHA256Hash,
+            toPath: destinationPath,
+            extraHeaders: extraHeaders,
+            allowPatch: false,
+            progressBlock: progressBlock,
+            successBlock: successBlock,
+            errorBlock: errorBlock
+          )
+          return
```

Comment:
same as android, maybe split these conditions to dedicated functions and add tests